protected/tests/functional_test/AdminSiteAccessTest.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * Functional test for only logged in admin user can access /site/admin.
+ */
+
+class AdminSiteAccessTest extends FunctionalTesting
+{
+    use BrowserSignInSteps;
+    use BrowserPageSteps;
+
+    public function testItShouldBeDisplayedToUsersWithAdminRole ()
+    {
+        // Logged in as Admin, and can visit to /site/admin page.
+        // 2 assertions, 2 passes.
+        $this->loginToWebSiteWithSessionAndCredentialsThenAssert("admin@gigadb.org","gigadb","Admin");
+        $url = "http://gigadb.dev/site/admin";
+        $this->visitPageWithSessionAndUrlThenAssertContentHasOrNull($url, "Administration Page");
+    }
+
+    public function testItShouldNotBeDisplayedToUsersWithUserRole ()
+    {
+        // Logged in as User, but cannot visit to /site/admin page.
+        // 2 assertions, login pass, but visit to /site/admin fail.
+        $this->loginToWebSiteWithSessionAndCredentialsThenAssert("user@gigadb.org","vagrant","User");
```

Comment:
I think it is more specific to check for `John's GigaDB Page` after the user has logged in. Also, confirm you have the correct password here too.